src/main/java/jarvis/model/Task.java

```diff
@@ -74,10 +79,15 @@ public boolean equals(Object other) {
             return false;
         }
 
+
         Task otherTask = (Task) other;
-        return otherTask.getDesc().equals(getDesc())
-                && otherTask.getDeadline().equals(getDeadline())
-                && otherTask.isDone() == isDone();
+
+        if (taskDeadline == null || otherTask.taskDeadline == null) {
+            return taskDeadline == null && otherTask.taskDeadline == null;
```

Comment:
This skips the other equality checks

Reply:
Thanks for reading the code. I realise I overlooked quite some stuff. They are fixed in my recent commit, let me know if there are more issues.